internal/envoy/v3/cluster_test.go

```diff
@@ -508,6 +508,22 @@ func TestCluster(t *testing.T) {
 				),
 			},
 		},
+		"cluster with connect timeout set": {
+			cluster: &dag.Cluster{
+				Upstream:       service(s1),
+				ConnectTimeout: 2 * time.Second,
```

Comment:
nit: since the test merges the want and default cluster config, maybe use a different value to ensure theres no chance a mistake in implementation still makes this test pass

Reply:
Good catch, changed to 10 seconds.